bokeh/core/property/regex.py

```diff
@@ -90,6 +92,23 @@ def validate(self, value, detail=True):
             msg = "" if not detail else "expected a string matching %r pattern, got %r" % (self.regex.pattern, value)
             raise ValueError(msg)
 
+class Base64String(String):
+    ''' Encode a ascii string using Base64.
+
+    Args:
+        value : a string to encode
+
+    Returns:
+        string
+
+    '''
+
+    def serialize_value(self, value):
+
+        if isinstance(value, str):
+            value = base64.b64encode(value.encode("utf-8"))
```

Comment:
So this will need one more `.decode("utf-8") at the end so that the properties system can serialize the base64 value (plain bytes can't be serialized). 

there should not be any changes needed on the JS side.